packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2171,18 +2172,24 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     }
 
     caretRect = caretRect.shift(_getPixelPerfectCursorOffset(caretRect));
+    _currentCaretRect = caretRect;
+
+    if (!_showCursor.value)
```

Comment:
It seems simpler to check `_showCursor.value` inside of this method like this instead of outside like it was before 👍 .